examples/query_demographics.js

```diff
@@ -22,86 +26,113 @@
-  camera_config.set_width(640);
-  camera_config.set_height(480);
-  config.malos_eye_config.set_camera_config(camera_config)
-  malosEyeConfigSocket.send(config.encode().toBuffer())
+
+  let camera = matrix_io.malos.v1.maloseye.CameraConfig.create({
+    cameraId: 0,
+    width: 640,
+    height: 480
+  });
+
+  let eye_config = matrix_io.malos.v1.maloseye.MalosEyeConfig.create({
+    cameraConfig: camera
+  });
+
+  let config = matrix_io.malos.v1.driver.DriverConfig.create({
+    delayBetweenUpdates: 0.05,
+    malosEyeConfig: eye_config
+  });
+
+  malosEyeConfigSocket.send(matrix_io.malos.v1.driver.DriverConfig.encode(config).finish())
 }
 
 ConfigureVideoCapture()
 
-function SetObjectsToDetect(objs) {
-  console.log('updating objects to detect')
-  var config = new matrixMalosBuilder.DriverConfig
-  config.malos_eye_config = new matrixMalosBuilder.MalosEyeConfig
-  for(var i = 0; i < objs.length; ++i) {
-    config.malos_eye_config.object_to_detect.push(objs[i])
-  }
-  malosEyeConfigSocket.send(config.encode().toBuffer())
+
+function ConfigureObjectsToDetect() {
+  let eye_config = matrix_io.malos.v1.maloseye.MalosEyeConfig.create({
+              objectToDetect: [matrix_io.malos.v1.maloseye.EnumMalosEyeDetectionType.FACE_DEMOGRAPHICS]
+            });
+  let  config = matrix_io.malos.v1.driver.DriverConfig.create({
+               malosEyeConfig : eye_config
+            });
+  malosEyeConfigSocket.send(matrix_io.malos.v1.driver.DriverConfig.encode(config).finish())
 }
 
-SetObjectsToDetect([matrixMalosBuilder.EnumMalosEyeDetectionType.FACE_DEMOGRAPHICS])
+ConfigureObjectsToDetect()
 
 // ********** End configuration.
 
+function roundpose(val) {
+  return Math.round(val * 1000) / 1000;
+}
+
 // ********** Start updates - Here is where they are received.
 var updateSocket = zmq.socket('sub')
 updateSocket.connect('tcp://' + creator_ip + ':' + (creator_demographics_base_port + 3))
 updateSocket.subscribe('')
 updateSocket.on('message', function(buffer) {
-    // .toRaw() gets you decoded values! Try what happens without it.
-    var data = new matrixVisionBuilder.VisionResult.decode(buffer).toRaw()
-    console.log(data)
-    for (var i = 0; i < data.rect_detection.length; ++i) {
-        console.log(data.rect_detection[i].facial_recognition)
+    let data = new matrix_io.vision.v1.VisionResult.decode(buffer)
+    for (let i = 0; i < data.rectDetection.length; ++i) {
+        console.log(' *** Detection ' + i + ' ***');
+        let detection = data.rectDetection[i];
+        console.log('Location: ', detection.location);
+        for (let j = 0; i < detection.facialRecognition.length; ++i) {
+          let rec = detection.facialRecognition[i];
+          if (rec.tag == 0) {
+            console.log('Age: ', rec.age);
+          } else if (rec.tag == 1) {
+            // TODO: Is there a way to do this conversion with protobufjs 6.X?
+            let emotion = 'unknown';
+            switch(rec.emotion) {
```

Comment:
I think there should be a way of using the constant names instead of the integers here, but it may involve a flag in the `matrix-protos` generation. You think it makes sense to adjust so we work with the enum names rather than the values ?

Reply:
Yes, they are. `matrix_io.malos.v1.io.GpioParams.EnumMode.OUTPUT` would be an example.